docs: add WebTransport docs (#9308)
ipfs/kubo: docs/changelogs/v0.16.md

```

### 🌍 WebTransport new experimental Transport

A new feature of [`go-libp2p`](https://github.com/libp2p/go-libp2p/releases/tag/v0.23.0) is [WebTransport](https://github.com/libp2p/go-libp2p/issues/1717).

For now it is **disabled by default** and considered **experimental**.
If you find issues running it please [report them to us](https://github.com/ipfs/kubo/issues/new).

> In the future we will change Kubo to listen on WebTransport by default (and listening) for anyone already listening `QUIC`, when we think it is stabilized.

Reviewer: ```suggestion
In the future Kubo will listen on WebTransport by default for anyone already listening on QUIC addresses.
```

Author: Done

> This transport will allow in-browser nodes to contact Kubo nodes using a new

Reviewer: can’t do multiline-suggestion from my tablet…
```suggestion
WebTransport is a new transport protocol currently under development by the IETF and the W3C, and already implemented by Chrome. Conceptually, it’s like WebSocket run over QUIC instead of TCP. Most importantly, it allows browsers to establish (secure!) connections to WebTransport servers without the need for CA-signed certificates, thereby enabling any js-libp2p node running in a browser to connect to any kubo node, with zero manual configuration involved.
```

Author: Done

fancy API which gets around mixed content issues, the end goal is for js-ipfs
running inside a browser to access the IPFS network as a client.

> #### How to enable WebTransport

Reviewer 2: Can you add these examples into the documentation too?

Author: Done

1. Enable the WebTransport transport:
   `ipfs config Swarm.Transports.Network.WebTransport --json true`
1. Add a listener address for WebTransport to your `Addresses.Swarm` key, for example:
   ```json
   [
     "/ip4/0.0.0.0/tcp/4001",
     "/ip4/0.0.0.0/udp/4001/quic",

>      "/ip4/0.0.0.0/udp/4002/quic/webtransport"

Reviewer 3: Is there a reason why a different port is used for WebTransport's address? Is it because the QUIC address technically takes it?

Reviewer 3: Nevermind, found a comment in config.md that mentions why. Maybe it should be mentioned here too? I think it'd trip people up if they only saw this example

Author: I've already asked the libp2p folks about that.
This is simply because no one wrote the code for the WebTransport transport and QUIC transport to talk together.

It might happen in the future, but this is pretty low priority.

Reviewer: Tracked in https://github.com/libp2p/go-libp2p/issues/1759. I'm actually planning to land this in the next release, so your migration for generating WebTransport addresses can be as simple as "take all the `/quic` addresses and add a `/webtransport`.

Author: Thx it's exactly what I was hopping to do.
